commands.py

```diff
@@ -452,41 +400,53 @@ async def join(self, ctx, *arg):
-        aliases = {"he": 747925204864335935, "she": 747925268181811211, "ze": 747925349232279552, "they": 747925313748729976}
+        aliases = {"he": "He/Him/His", "she": "She/Her/Hers", "ze": "Ze/Zir/Zirs", "they": "They/Them/Theirs"}
 
-        for role in ctx.guild.roles:
-            if name == role.name.lower() and not role.name.startswith("221dm-"):
-                if role.name not in valid_roles:
-                    self.add_instructor_role_counter += 1
-                    if self.add_instructor_role_counter > 5:
-                        if self.add_instructor_role_counter == 42:
-                            if random.random() > 0.999:
-                                return await ctx.send("Congratulations, you found the secret message. IDEK how you did it, but good job. Still can't add the instructor role though. Bummer, I know.", delete_after=5)
-                        elif self.add_instructor_role_counter == 69:
-                            if random.random() > 0.9999:
-                                return await ctx.send("nice.", delete_after=5)
-                        return await ctx.send("You can't add that role, but if you try again, maybe something different will happen on the 42nd attempt", delete_after=5)
-                    else:
-                        return await ctx.send("you cannot add an instructor/invalid role!", delete_after=5)
-                elif name.startswith("l1") and any(role.name.startswith("L1") for role in ctx.author.roles):
-                    return await ctx.send("you already have a lab role!", delete_after=5)
-                elif role in ctx.author.roles:
-                    return await ctx.send("you already have that role!", delete_after=5)
-                else:
-                    await ctx.author.add_roles(role)
-                    return await ctx.send("role added!", delete_after=5)
-            elif name in aliases:
-                await ctx.author.add_roles(ctx.guild.get_role(aliases[name]))
-                return await ctx.send("role added!", delete_after=5)
-        else:
-            await ctx.send("you can't add that role!", delete_after=5)
+        # Convert alias to proper name
+        if name in aliases:
+            name = aliases[name].lower()
+
+        # Ensure that people only add one lab role
+        if name.startswith("l1") and any(role.name.startswith("L1") for role in ctx.author.roles):
+            return await ctx.send("You already have a lab role!", delete_after=5)
+
+        # Grab the rolee that the user selected
+        role = next((r for r in ctx.guild.roles if name == r.name.lower()), None)
+
+        # Check that the role actually exists
+        if not role:
+            await ctx.send("You can't add that role!", delete_after=5)
             return await ctx.send(ctx.command.help)
```

Comment:
hmmmm what if we raise an exception, and then do the exception catching in cs221bot.py, where we can send ctx.command.help in one place, instead of having to put that at literally every fail state